module/VuFindSearch/src/VuFindSearch/Backend/EDS/SearchRequestModel.php

```diff
@@ -517,4 +547,46 @@ public function __set($property, $value)
 
         return $this;
     }
+
+    // all copied for draft purposes from Base/Params
+    protected $orFacets = [];
+
+    // protected function initFacetList($facetList, $facetSettings, $cfgFile = null)
+    public function initFacetList($facetList, $facetSettings, $config)
+    {
+        // $config = $this->configLoader
+        //     ->get($cfgFile ?? $this->getOptions()->getFacetsIni());
+        if (!isset($config->$facetList)) {
+            return false;
+        }
+        if (isset($config->$facetSettings->orFacets)) {
+            $orFields
+                = array_map('trim', explode(',', $config->$facetSettings->orFacets));
+        } else {
+            $orFields = [];
+        }
+        foreach ($config->$facetList as $key => $value) {
+            $useOr = (isset($orFields[0]) && $orFields[0] == '*')
+                || in_array($key, $orFields);
+            $this->addFacet($key, $value, $useOr);
+        }
+
+        return true;
+    }
+
+    public function addFacet($newField, $newAlias = null, $ored = false)
+    {
+        // if ($newAlias == null) {
+        //     $newAlias = $newField;
+        // }
+        // $this->facetConfig[$newField] = $newAlias;
+        if ($ored) {
+            $this->orFacets[] = $newField;
+        }
+    }
+
+    public function getFacetOperator($field)
+    {
+        return in_array($field, $this->orFacets) ? 'OR' : 'AND';
+    }
```

Comment:
So obviously I don't really want to duplicate all of this logic from Base/Params. But I could not find a clean way to make the Params object available to SearchRequestModel. Hoping for some advice here.

Reply:
It's been quite a while since I've looked at the EDS code, so maybe there's an obvious answer to this question... but why do you want to have this logic in the SearchRequestModel at all? I think the way this is supposed to work is that the Params class constructs a ParamBag which gets sent to the backend, and then the backend turns the ParamBag into a SearchRequestModel for the EDS code to work with. Is there a reason we can't do all the processing directly in Params and then just pass the processed data all the way through? Or are there code paths that go straight to the backend without being processed by Params first?

(I can take a deeper look at the code if you need me to -- but that's my first impression based on memory and no code reading :-) ).

Reply:
Well, all this (duplicate) processing is just so that I can call `getFacetOperator` here in SearchRequestModel, which is necessary to build the (completely different) structure for AND vs OR.  Which, I don't like.  With your point about keeping the logic in EDS/Params, I'm thinking I should change AbstractEDSParams->createBackendFilterParams() to generate filters that look like `field:operator:value` instead of `field:value`.  And then parse the operator in SearchRequestModel->addfilter().  I would still have to store that AND/OR state internally somewhere in SearchRequestModel until I actually format the EBSCO calls, but that's an improvement.  Let me know if that sounds way off, if not I will build it.

Reply:
Yes, that's exactly the sort of thing I had in mind -- this code is only ever used for EDS, and its purpose is to map the VuFind standard structures to something useful for EDS, so I think it makes perfect sense to pass along any and all needed data at this stage in order to simplify later steps.

If it's not overly difficult, you should probably aim to build things in a backward-compatible way -- e.g. if you're adding `field:operator:value` format, the subsequent code should still handle `field:value` format by inserting a default operator value -- just in case somebody has written custom code that talks directly to the EDS backend. I don't think it's very likely, but if we want to include this in 10.1 instead of 11.0, we should make a greater effort to avoid breaking things. :-)